zeppelin-interpreter/src/main/java/org/apache/zeppelin/scheduler/RemoteScheduler.java

```diff
@@ -167,14 +166,15 @@ public void setMaxConcurrency(int maxConcurrency) {
     private long initialPeriodMsec;
     private long initialPeriodCheckIntervalMsec;
     private long checkIntervalMsec;
-    private boolean terminate;
+    private volatile boolean terminate;
```

Comment:
This is because terminate is used by multiple thread, it is much safer to make it `volatile`.  Otherwise it zeppelin server may detect the job as terminated later than the correct timestamp.